### ISSUE TITLE
App shows different charge times on 22 KW than shown on chargeprice.app (affects calculated prices)

### ISSUE THREAD

**Issue by cremor** [johan12345/EVMap#170]
I noticed that EVMap sometimes shows charge times (and therefore prices for time based tariffs) that are very different from the ones shown on [chargeprice.app](https://www.chargeprice.app).

Settings:
Car: Hyundai Ioniq (2016)
Charge setting: 20% to 80%

For a 22 kW type 2 port EVMap shows a charge time of 4:07 h.
But chargeprice.app shows a charge time of 2:50 h (18 kWh with 6.6 kW average)

I think the value shown on chargeprice.app is the correct one. This car can charge with a maximum of 6.6 kW with 32 A over one phase AC. A 22 kW charge port provides 32 A, so this maximum can be reached.
In comparison, a 11 kW charge port only provides 16 A, so the charge time should be about twice as long. Both EVMap and chargeprice.app show a charge time of 5:00 h for 11 kW charge ports. So 2:50 h for 22 kW looks way more correct than 4:07 h.

This only seems to affect 22 kW charge ports. For all other charge speeds that I checked the values shown in EVMap are the same as shown on chargeprice.app.

**Comment by johan12345** [comment]
Hmm, I think this is due to the option "Allow unbalanced load" ("Schieflast erlauben") on Chargeprice. If not enabled, it limits single-phase charging to 4.5 kW (20 A). This option is disabled by default both on the Chargeprice website and in the API, and EVMap does not yet have a way to change that. Probably you had the option enabled in Chargeprice and this is why you are seeing different times and prices.

Many 22 kW public chargers detect unbalanced loads and limit the current to 20 A, so the longer duration may actually be correct (see e.g. [here](https://www.goingelectric.de/forum/viewtopic.php?f=119&t=70112)). But yes, I can add an option to change this setting.

**Comment by cremor** [comment]
You are right, it's that option.

TBH I have not tried it yet how fast the car will actually charge on such a port. But I found a charge log entry on goingelectric.de where someone with an Hyundai Ioniq reported that he could charge with 7 kW (I assume he rounded 6.6 kW) on a 22 kW port. This is in Austria, maybe the rules/detections work different here than in Germany?

**Comment by johan12345** [comment]
Yeah, could be that the regulations are different in Austria. I'm also not completely sure how common this limitation is for public chargers in Germany, as I have never driven a single-phase charging car.
